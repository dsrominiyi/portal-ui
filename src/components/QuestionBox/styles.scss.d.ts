export const questionBox: string;
export const hasError: string;
export const questionText: string;
export const row: string;
export const errorText: string;
