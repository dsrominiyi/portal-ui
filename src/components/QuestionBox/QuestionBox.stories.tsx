import React, { ReactNode } from 'react';
import { storiesOf } from '@storybook/react';

import QuestionBox from '.';
import { storyRenders as textFieldStoryRender } from '../TextField/TextField.stories';
import { storyRenders as dropdownStoryRender } from '../Dropdown/Dropdown.stories';
import { storyRenders as affixFieldStoryRender } from '../TextField/AffixedField/AffixedField.stories';

const stories = storiesOf('QuestionBox', module);

const render = (
  inputFields: ReactNode[],
  text?: string,
  hasError?: boolean,
  errorText?: string
) => (props?: any) => (
  <QuestionBox
    key={props && props.key}
    text={text}
    inputFields={inputFields}
    hasError={hasError}
    errorText={errorText}
  />
);

const componentKey = { key: 0 };
const textField = textFieldStoryRender.emailField(componentKey);
const dropdown = dropdownStoryRender.withSelectedValue(componentKey);
const affixedField = affixFieldStoryRender.prefixed(componentKey);

export const storyRenders = {
  withTextField: render([textField], 'With TextField'),
  withDropdown: render([dropdown], 'With Dropdown'),
  withAffixedField: render([affixedField], 'With Affixed Field'),
  withoutQuestionText: render([textField]),
  withErrorAndText: render([textField], undefined, true, 'Please confirm first name')
};

stories.add('With TextField', storyRenders.withTextField);
stories.add('With Dropdown', storyRenders.withDropdown);
stories.add('With AffixedField component', storyRenders.withAffixedField);
stories.add('Without question text', storyRenders.withoutQuestionText);
stories.add('With error and text', storyRenders.withErrorAndText);
