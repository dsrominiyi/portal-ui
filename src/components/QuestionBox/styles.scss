@import '../../style/vars';

.question-box {
  display: flex;
  flex-direction: column;
  border-bottom: 1px solid $brand-grey-white;
  padding: 20px 20px 20px 16px;
  border-left: 4px solid transparent;
  background-color: $brand-white-smoke;
  transition: border 0.2s;

  &.has-error {
    border-left: 4px solid $brand-danger;
  }
}

.question-text {
  margin-bottom: 1.25rem;
}

.row {
  display: flex;
  width: 100%;
}

.error-text {
  margin-top: 1.25rem;
  font-size: 1rem;
  font-weight: bold;
  color: $brand-danger;
}
