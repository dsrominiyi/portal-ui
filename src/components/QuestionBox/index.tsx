import React, { FunctionComponent, ReactNode } from 'react';
import classnames from 'classnames/bind';

import formStyles from '../../forms/styles.scss';
import styles from './styles.scss';

interface Props {
  additionalClasses?: string[];
  text?: string;
  inputFields: ReactNode[];
  hasError?: boolean;
  errorText?: string;
}

const cx = classnames.bind(styles);

export const QuestionBox: FunctionComponent<Props> = ({
  additionalClasses = [],
  text,
  inputFields,
  hasError,
  errorText
}: Props) => (
  <div className={[...additionalClasses, cx({ questionBox: true, hasError })].join(' ')}>
    {text && <div className={[formStyles.label, styles.questionText].join(' ')}>{text}</div>}
    <div>{inputFields}</div>
    {hasError && errorText && <div className={styles.errorText}>{errorText}</div>}
  </div>
);

QuestionBox.displayName = 'QuestionBox';

export default QuestionBox;
