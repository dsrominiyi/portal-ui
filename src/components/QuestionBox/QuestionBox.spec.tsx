import { storyRenders } from './QuestionBox.stories';

import testStorySnapshots from '../../helpers/test/testStorySnapshots';

describe('QuestionBox', () => {
  test('component story renders match snapshots', () => {
    testStorySnapshots(storyRenders);
  });
});
