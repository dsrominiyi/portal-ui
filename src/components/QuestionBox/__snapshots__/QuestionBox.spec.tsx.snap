// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionBox component story renders match snapshots: withAffixedField 1`] = `
<div
  className="questionBox"
>
  <div
    className="label questionText"
  >
    With Affixed Field
  </div>
  <div>
    <AffixedField
      key="0"
      onChange={[Function]}
      prefix="Â£"
    />
  </div>
</div>
`;

exports[`QuestionBox component story renders match snapshots: withDropdown 1`] = `
<div
  className="questionBox"
>
  <div
    className="label questionText"
  >
    With Dropdown
  </div>
  <div>
    <Dropdown
      key="0"
      onChange={[Function]}
      options={
        Array [
          Object {
            "text": "1",
            "value": "value1",
          },
          Object {
            "text": "2",
            "value": "value2",
          },
          Object {
            "text": "3",
            "value": "value3",
          },
          Object {
            "text": "4",
            "value": "value4",
          },
        ]
      }
      selected="key2"
    />
  </div>
</div>
`;

exports[`QuestionBox component story renders match snapshots: withErrorAndText 1`] = `
<div
  className="questionBox hasError"
>
  <div>
    <TextField
      icon={<SvgUser />}
      key="0"
      label="Email Address"
      onChange={[Function]}
      password={false}
      value="test@datamatters.co.uk"
    />
  </div>
  <div
    className="errorText"
  >
    Please confirm first name
  </div>
</div>
`;

exports[`QuestionBox component story renders match snapshots: withTextField 1`] = `
<div
  className="questionBox"
>
  <div
    className="label questionText"
  >
    With TextField
  </div>
  <div>
    <TextField
      icon={<SvgUser />}
      key="0"
      label="Email Address"
      onChange={[Function]}
      password={false}
      value="test@datamatters.co.uk"
    />
  </div>
</div>
`;

exports[`QuestionBox component story renders match snapshots: withoutQuestionText 1`] = `
<div
  className="questionBox"
>
  <div>
    <TextField
      icon={<SvgUser />}
      key="0"
      label="Email Address"
      onChange={[Function]}
      password={false}
      value="test@datamatters.co.uk"
    />
  </div>
</div>
`;
